src/main/flight/pid.c

```diff
@@ -978,6 +983,57 @@ static void pidApplyCyclicMode3(uint8_t axis, const pidProfile_t * pidProfile)
 
     // Calculate Offset component
     pid.data[axis].axisOffset = limitf(pid.data[axis].axisOffset + offDelta, pid.offsetLimit[axis]);
+
+    // Offset flood: convert axisError to axisOffset
+    // The algorithm only makes sense if both Ki and Ko !=0;
+    if (pid.coef[axis].Ki != 0 && pid.coef[axis].Ko != 0) {
+        const float axisError = pid.data[axis].axisError;
+        const float axisOffset = pid.data[axis].axisOffset;
+        const float Ki = pid.coef[axis].Ki;
+        const float Ko = pid.coef[axis].Ko;
+
+        // 0. calculate bleed rate
+        float bleedRate = pidTableLookup(curve, pidProfile->offset_flood_curve,
+                                         LOOKUP_CURVE_POINTS) *
+                          0.08f;
+        bleedRate = copysignf(bleedRate, axisError);
+        bleedRate *= pid.offsetFloodRelaxFactor;
+
+        // 1. offsetDelta = value to be added to axisOffset
+        float offsetDelta = bleedRate * pid.dT;
+        // 1. determin sign of offsetDelta
+        // offsetDelta is positive if bleedRate>0 && collective>0 || bleedRate<0
+        // && collective<0
+        offsetDelta = copysignf(offsetDelta, bleedRate * collective);
+
+        // 1. Check offsetLimit
+        offsetDelta = limitf(axisOffset + offsetDelta, pid.offsetLimit[axis]) -
+                      axisOffset;
+
+        // 2. calculate equivalent output delta and errorDelta
+        // errorDelta = value to be substract from axisError.
+        // Note:
+        //    output = axisOffset * collective * Ko
+        //    output = axisError * Ki
+        float outputDelta = collective * offsetDelta * Ko;
+        float errorDelta = outputDelta / Ki;
+
+        // 2. Check axisError limit
+        // Note: axisError and errorDelta have same sign
+        // collective == 0 -> errorDelta == 0 and we will not enter (safe)
+        if (fabsf(axisError) - fabsf(errorDelta) < 0) {
+            // We need to re-calculate outputDelta and offsetDelta:
+            errorDelta = axisError;
+            outputDelta = errorDelta * Ki;
+            offsetDelta = outputDelta / collective / Ko;
+        }
+
+        // 3. Update axisError and axisOffset
+        pid.data[axis].axisError -= errorDelta;
+        pid.data[axis].I -= errorDelta * Ki;
+        pid.data[axis].axisOffset += offsetDelta;
+    }
+
```

Comment:
Move this block into a separate function pidApplyOffsetFlood(), just like pidApplyOffsetBleed()

Reply:
Done